argm.c

```diff
@@ -311,17 +311,25 @@ argm_combine_universal(PG_FUNCTION_ARGS, int32 compareFunctionResultToAdvance)
 
     GET_AGG_CONTEXT(fcinfo, aggcontext);
 
-    if (PG_ARGISNULL(1))
-        PG_RETURN_POINTER(PG_GETARG_POINTER(0));
+    state0 = PG_ARGISNULL(0) ? NULL : (ArgmState *) PG_GETARG_POINTER(0);
+    state1 = PG_ARGISNULL(1) ? NULL : (ArgmState *) PG_GETARG_POINTER(1);
+
+    if (state1 == NULL) {
+        /* just pass state0 down the line */
+        if (state0 == NULL)
+            PG_RETURN_NULL();
```

Comment:
And this is the fix for the bug point A: we return SQL null, if state0 was SQL null.